pkg/ingress/controller/controller.go

```diff
@@ -824,10 +825,22 @@ func (c *Controller) ensureIngress(ing *nwv1.Ingress) error {
 			var policyRules []l7policies.CreateRuleOpts
 
 			if host != "" {
+
+				hostRegex := "^"
```

Comment:
I think we need a comment to describe (maybe links as well) to describe why we do like this, maybe sample as well?

Reply:
Links to:
* Octavia rule creation API: https://docs.openstack.org/api-ref/load-balancer/v2/?expanded=create-an-l7-rule-detail#create-an-l7-rule
* Ingress Rule.Host: https://github.com/kubernetes/kubernetes/blob/189fe3f3e6d157bc63c2276f79c3de860d7d8c57/pkg/apis/networking/types.go#L444-L467C2

This would also be super simple to unit test if we pulled it into a small separate function, and it's probably complicated enough to deserve it.